app/src/main/java/de/danoeh/antennapod/fragment/AudioPlayerFragment.java

```diff
@@ -386,6 +390,24 @@ public void updatePosition(PlaybackPositionEvent event) {
         }
         float progress = ((float) event.getPosition()) / event.getDuration();
         sbPosition.setProgress((int) (progress * sbPosition.getMax()));
+        Playable media = controller.getMedia();
+        if (media instanceof FeedMedia)  {
+            FeedMedia feedMedia = (FeedMedia) media;
+            FeedPreferences preferences = feedMedia.getItem().getFeed().getPreferences();
+            int skipEnd = preferences.getFeedSkipEnding();
+            if (skipEnd > 0 && remainingTime < skipEnd * 1000) {
+                Log.d(TAG, "Skipping the remaining " + remainingTime / 1000 + " " + skipEnd );
+                String skipMesg = getActivity().getString(de.danoeh.antennapod.core.R.string.pref_feed_skip_ending) + " " +
+                        skipEnd + " " +
+                        getActivity().getString(de.danoeh.antennapod.core.R.string.time_seconds);
+                Toast toast = Toast.makeText(getActivity(), skipMesg,
+                        Toast.LENGTH_LONG);
+                toast.show();
+                IntentUtils.sendLocalBroadcast(getActivity(), PlaybackService.ACTION_SKIP_CURRENT_EPISODE);
```

Comment:
The UI should not be responsible for doing playback actions. If the user closes the UI, episodes should still get skipped. When adding the skipping code like this and swiping out the player during playback, this is no longer the case.

Additionally, this introduces code duplication. This should be done centrally, somewhere in the PlaybackService.

Reply:
done.